lib/modules/sync_seeds.rb

```diff
@@ -0,0 +1,137 @@
+  end
+  
+  def create_paths(relative_path)
+    { local_path: File.join(LOCAL, relative_path), remote_path: File.join(REMOTE, relative_path) }
+  end
+
+  def check_if_newer(parent_folder, local_item, remote_item, remote_path, local_path, base = LOCAL)
+    if parent_folder.include?(local_item)
+      yield if block_given?
+      
+      is_newer = Time.at(remote_item.attributes.mtime) >= File.stat(local_path).mtime  
+      # If there are any outdated files, will trigger download
+      if is_newer
+        if Dir.glob('*', base: LOCAL).include?(local_item)
+          download_files(remote_path, local_path) 
+          # Will be hit if local_item is a file or folder inside a directory
+        elsif Dir.glob('**/*', base: base).include?(local_item)
+          download_files(remote_path, local_path)
+          downloaded = true  
+        end              
+      end
+    else
+      # Just download it if it doesn't exist at all
+      download_files(remote_path, LOCAL)
+      downloaded = true
+    end
+
+    downloaded
+  end
+
+  def compare_folders(wildcard, local, remote, base = LOCAL)
+    puts "Checking to see what files need to be deleted from #{base}" 
+
+    remote_list = @session.sftp.dir.glob(remote, wildcard).map do |f|  
+      f.name.force_encoding('UTF-8')
+    end
+
+    local_list = Dir.glob(wildcard, base: local)
+
+    files_for_deletion(local_list, remote_list, base)
+  end
+
+  def check_inside_folder(folder, local_list)
+    paths = create_paths(folder)
+
+    compare_folders('**/*', paths[:local_path], paths[:remote_path], paths[:local_path])
+
+    local_folder_content = Dir.glob('**/*', base: paths[:local_path])
+    remote_folder_content = @session.sftp.dir.glob(paths[:remote_path], '**/*')
+    
+    # We don't want to download the whole folder again if it's already been re-downloaded once
+
+
+    remote_folder_content.each do |file|
+      # Go through the various files and folders and check to see if they exist locally
+      local_file = file.name.force_encoding('UTF-8')
+
+      check = check_if_newer(local_folder_content, local_file, file, paths[:remote_path], paths[:local_path], paths[:local_path])
+  
+      # Break out of loop if already downloaded
+      break if check == true
+    end
+  end
+
+  def main_task(local_list, remote_list)
+    remote_list.each do |object|
+      # There are files with non-ASCII characters (i.e. accented) in the CMS files
+      name = object.name.force_encoding('UTF-8')
+      paths = create_paths(name)
+      
+      
+      check_if_newer(local_list, name, object, paths[:remote_path], paths[:local_path]) do 
+        check_inside_folder(name, local_list) if object.attributes.directory?
+      end
+    end
+  end
+
+  def commence_import(answer)
+    logger = ComfortableMexicanSofa.logger
+    ComfortableMexicanSofa.logger = Logger.new(STDOUT)
+
+    if answer == 'All'
+      Rake::Task["comfy:cms_seeds:import"].invoke('protected-planet', 'protectedplanet')     
```

Comment:
I feel like it might be useful to have the folder name and site name be passed as arguments to the task. This is so you can change this when required at the moment of running the task without changing the code. This would also affect the `LOCAL` constant in the rake task 

Worth also adding a description/comment above the task on how to run if after the changes (like you just did for the seed_type)

Reply:
So my rake task would be invoked like the Comfy task with 'protected-planet', 'protectedplanet'?

Reply:
Potentially yeah. Or you could check whether the folder name would be enough on it's own, meaning that at the moment of importing the folder name should be exactly the same as the site name on the CMS